src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6116,6 +6116,26 @@ bool VM_Crac::is_claimed_fd(int fd) {
   return false;
 }
 
+static void wakeup_threads_in_timedwait() {
+  SafeThreadsListPtr listPtr(Thread::current(), true);
+  ThreadsList *list = listPtr.list();
+  assert(list != NULL, "Thread list is null");
+  for (uint i = 0; i < list->length(); ++i) {
+    JavaThread* t = list->thread_at(i);
```

Comment:
Please use Threads::threads_do, that includes non-Java threads as well.

Reply:
Good point, I'll use `Threads::java_threads_do`. Not sure what I could do with non-Java threads, I haven't found any use of `pthread_cond_timedwait` from non-Java threads except the WatcherThread (and we can't fix non-JVM threads, we need to track any condition to signal).